package_managers/download_pkgs.bzl

```diff
@@ -16,61 +16,65 @@
 
 load("//package_managers/apt_get:apt_get.bzl", "generate_apt_get")
 load("//package_managers/apt_get:repos.bzl", "generate_additional_repos")
+load("//package_managers:package_manager_provider.bzl", "package_manager_provider")
 load("@io_bazel_rules_docker//docker:docker.bzl", "docker_build")
 
 def _impl(ctx):
+    package_manager = ctx.attr.package_manager_generator[package_manager_provider]
     # docker_build rules always generate an image named 'bazel/$package:$name'.
     builder_image_name = "bazel/%s:%s" % (ctx.attr.image_tar.label.package,
                                           ctx.attr.image_tar.label.name.split(".tar")[0])
 
     # Generate a shell script to run apt_get inside this docker image.
     # TODO(tejaldesai): Replace this by docker_run rule
     build_contents = """\
+# autogenerated - do not edit.
 #!/bin/bash
 set -ex
 docker load --input {image_tar}
 # Run the builder image.
-cid=$(docker run -d --privileged {image_name} /bin/bash)
+cid=$(docker run -d --privileged {image_name} sh -c $'{download_commands}')
 docker attach $cid
 docker cp $cid:{installables}.tar {output}.tar
 # Cleanup
 docker rm $cid
- """.format(image_tar="{0}/{1}".format(ctx.label.package,ctx.attr.image_tar.label.name),
+ """.format(image_tar=ctx.file.image_tar.short_path,
             image_name=builder_image_name,
             installables=ctx.attr.package_manager_generator.label.name,
-            output="{0}/{1}".format(ctx.label.package, ctx.attr.name))
+            download_commands=' && '.join(package_manager.download_commands),
```

Comment:
Do we need to join with &&? Can we join with newlines instead, since we're using sh -c?

Reply:
will give it a try.

Reply:
done